### PR TITLE
Fix a few syntax issues in shell scripts

### PR DESCRIPTION
**Pull request by am11**
This delta updates shell scripts for consistency. In particular, avoid mixing Bash and Unix shell syntax by replacing:
* `[ $x == $y ]` and `[[ $x = $y ]]` with `[[ $x == $y ]]`
* `[[ $x ]] && [[ $y ]]` and `[ $x ] && [ $y ]` with `[[ $x && $y ]]`
*  `[ ! -z $x ]` and `[ $x != "" ]` with `[ -n $x ]`
*  `[ ! -n $x ]` and `[  $x == "" ]` with `[ -z $x ]`
* `! $x == $y` with `$x != $y`
* `#!/bin/bash` with `#!/usr/bin/env bash`
* `[[` with `[` in files with `/bin/sh` shebang

### CODE REVIEW

File: src/libraries/System.Net.Http/tests/StressTests/HttpStress/run-docker-compose.sh [hunk 1]
```diff
@@ -67,23 +67,23 @@ done
 
 repo_root=$(git rev-parse --show-toplevel)
 
-if [[ buildcurrentlibraries -eq 1 ]]; then
+if [[ "$buildcurrentlibraries" -eq 1 ]]; then
```
Comment (am11) [inline src/libraries/System.Net.Http/tests/StressTests/HttpStress/run-docker-compose.sh:70]: This was a bug (missing `$`).

File: src/mono/mono/tests/verifier/make_il_overflow_test.sh [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #! /bin/sh
 
 SED="sed"
-if [ `which gsed 2> /dev/null` ] ; then 
+if [ `which gsed 2> /dev/null` ] ; then
```
Comment (hoyosjs) [inline src/mono/mono/tests/verifier/make_il_overflow_test.sh:4]:
```suggestion
if [ `which gsed 2> /dev/null` ]; then
```

File: src/tests/BuildWasmApps/Wasm.Build.Tests/data/RunScriptTemplate.sh [hunk 3]
```diff
@@ -4,13 +4,13 @@ EXECUTION_DIR=$(dirname $0)
...
 
-if [ -z "$HELIX_WORKITEM_UPLOAD_ROOT" ]; then
+if [[ -z "$HELIX_WORKITEM_UPLOAD_ROOT" ]]; then
 	XHARNESS_OUT="$EXECUTION_DIR/xharness-output"
 else
 	XHARNESS_OUT="$HELIX_WORKITEM_UPLOAD_ROOT/xharness-output"
 fi
 
-if [ ! -z "$XHARNESS_CLI_PATH" ]; then
+if [[ -n "$XHARNESS_CLI_PATH" ]]; then
```
Comment (hoyosjs) [inline src/tests/BuildWasmApps/Wasm.Build.Tests/data/RunScriptTemplate.sh:13]: Any reason to use [[ when most places changes use [

**Review by hoyosjs — COMMENTED** [review]
Just a couple minor questions.

File: eng/run-test.sh [hunk 4]
```diff
@@ -288,7 +283,7 @@ else
     platform="$(uname)"
     if [ "$platform" = "FreeBSD" ]; then
       maxProcesses=$(($(sysctl -n hw.ncpu)+1))
-    if [ "$platform" = "NetBSD" ] || [ "$platform" = "SunOS" ] ; then
+    if [ "$platform" = "NetBSD" ] || [ "$platform" = "SunOS" ]; then
```
[the remark this answers is not in the document]
Comment (am11) [reply eng/run-test.sh]: Nope, it was an oversight. Thanks for the catch! :)

File: src/mono/mono/tests/verifier/make_il_overflow_test.sh (hunk 2 above)
Comment (am11) [reply src/mono/mono/tests/verifier/make_il_overflow_test.sh]: Changed all occurrences of `] ;` with `];`.

File: src/tests/BuildWasmApps/Wasm.Build.Tests/data/RunScriptTemplate.sh (hunk 3 above)
Comment (am11) [reply src/tests/BuildWasmApps/Wasm.Build.Tests/data/RunScriptTemplate.sh:13]: It was discussed once with @janvorli and the consensus was to prefer `[[` over `[` in bash context: https://github.com/dotnet/runtime/pull/1753#discussion_r368626175.

I haven't changed all instances of `[[` -> `[` in bash scripts exhaustively in this PR, but mainly focused on those where we were mixing bash and unix shell syntax in bash scripts: `[[ $x = $y ]]` and `[ $x == $y ]`.

**Review by hoyosjs — APPROVED** [review]
LGTM. Thanks!

**Comment by am11** [comment]
Resolved merge conflict with https://github.com/dotnet/runtime/commit/85642f8147d9e2ba860c21baea25cbb4f12e665e.
@hoyosjs, I don't have permissions, could you please merge this? Thanks.

**Review by safern — APPROVED** [review]

**Comment by safern** [comment]
@radical could you please look at the staging failure?

**Comment by antonfirsov** [comment]
This change broke the (non-mandatory) SslStress pipeline.

When touching `SslStress` and `HttpStress`, please make sure to run the corresponding pipelines:

```
/azp run runtime-libraries stress-http
/azp run runtime-libraries stress-ssl
```

**Pull request merged**